Traceback (most recent call last):
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/console/logging/io_handler.py", line 22, in emit
    msg = self.format(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 943, in format
    return fmt.format(record)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/console/logging/io_formatter.py", line 39, in format
    formatted = super().format(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 678, in format
    record.message = record.getMessage()
  File "/usr/local/lib/python3.10/logging/__init__.py", line 368, in getMessage
    msg = msg % self.args
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/py_info.py", line 270, in __str__
    (
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/py_info.py", line 297, in spec
    # this method is not used by itself, so here and called functions can import stuff locally
AttributeError: 'PythonInfo' object has no attribute 'free_threaded'
Call stack:
  File "/usr/local/lib/python3.10/threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 83, in _worker
    work_item.run()
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 274, in _execute_operation
    result = self._do_execute_operation(operation)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 385, in _do_execute_operation
    result: int = getattr(self, f"_execute_{method}")(operation)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 509, in _execute_install
    status_code = self._install(operation)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 543, in _install
    archive = self._download_link(operation, Link(package.source_url))
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 714, in _download_link
    archive = self._chef.prepare(archive, output_dir=original_archive.parent)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/chef.py", line 43, in prepare
    return self._prepare_sdist(archive, destination=output_dir)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/chef.py", line 88, in _prepare_sdist
    return self._prepare(
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/chef.py", line 49, in _prepare
    with isolated_builder(
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/utils/isolated_build.py", line 180, in isolated_builder
    with ephemeral_environment(
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/utils/env/__init__.py", line 47, in ephemeral_environment
    EnvManager.build_venv(
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/utils/env/env_manager.py", line 552, in build_venv
    cli_result = virtualenv.cli_run(args, setup_logging=False)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/run/__init__.py", line 31, in cli_run
    of_session = session_via_cli(args, options, setup_logging, env)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/run/__init__.py", line 49, in session_via_cli
    parser, elements = build_parser(args, options, setup_logging, env)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/run/__init__.py", line 77, in build_parser
    parser._interpreter = interpreter = discover.interpreter  # noqa: SLF001
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/discover.py", line 41, in interpreter
    self._interpreter = self.run()
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py", line 60, in run
    result = get_interpreter(python_spec, self.try_first_with, self.app_data, self._env)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py", line 81, in get_interpreter
    LOGGER.info("proposed %s", interpreter)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1477, in info
    self._log(INFO, msg, args, **kwargs)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1624, in _log
    self.handle(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1634, in handle
    self.callHandlers(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1696, in callHandlers
    hdlr.handle(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 968, in handle
    self.emit(record)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/console/logging/io_handler.py", line 30, in emit
    self.handleError(record)
Message: '<fg=blue>proposed %s</>'
Arguments: (PythonInfo({'platform': 'linux', 'implementation': 'CPython', 'version_info': VersionInfo(major=3, minor=10, micro=16, releaselevel='final', serial=0), 'architecture': 64, 'version_nodot': '310', 'version': '3.10.16 (main, Jan 14 2025, 02:25:17) [GCC 12.2.0]', 'os': 'posix', 'prefix': '/usr/local', 'base_prefix': '/usr/local', 'real_prefix': None, 'base_exec_prefix': '/usr/local', 'exec_prefix': '/usr/local', 'executable': '/usr/local/bin/python3.10', 'original_executable': '/usr/local/bin/python3.10', 'system_executable': '/usr/local/bin/python3.10', 'has_venv': True, 'path': ['/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery', '/usr/local/lib/python310.zip', '/usr/local/lib/python3.10', '/usr/local/lib/python3.10/lib-dynload', '/usr/local/lib/python3.10/site-packages'], 'file_system_encoding': 'utf-8', 'stdout_encoding': 'utf-8', 'sysconfig_scheme': None, 'sysconfig_paths': {'stdlib': '{installed_base}/{platlibdir}/python{py_version_short}', 'platstdlib': '{platbase}/{platlibdir}/python{py_version_short}', 'purelib': '{base}/lib/python{py_version_short}/site-packages', 'platlib': '{platbase}/{platlibdir}/python{py_version_short}/site-packages', 'include': '{installed_base}/include/python{py_version_short}{abiflags}', 'scripts': '{base}/bin', 'data': '{base}'}, 'distutils_install': {'purelib': 'lib/python3.10/site-packages', 'platlib': 'lib/python3.10/site-packages', 'headers': 'include/python3.10/UNKNOWN', 'scripts': 'bin', 'data': ''}, 'sysconfig': {'makefile_filename': '/usr/local/lib/python3.10/config-3.10-x86_64-linux-gnu/Makefile'}, 'sysconfig_vars': {'platbase': '/usr/local', 'platlibdir': 'lib', 'py_version_short': '3.10', 'base': '/usr/local', 'PYTHONFRAMEWORK': '', 'abiflags': '', 'installed_base': '/usr/local'}, 'system_stdlib': '/usr/local/lib/python3.10', 'system_stdlib_platform': '/usr/local/lib/python3.10', 'max_size': 9223372036854775807}),)
--- Logging error ---
Traceback (most recent call last):
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/console/logging/io_handler.py", line 22, in emit
    msg = self.format(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 943, in format
    return fmt.format(record)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/console/logging/io_formatter.py", line 39, in format
    formatted = super().format(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 678, in format
    record.message = record.getMessage()
  File "/usr/local/lib/python3.10/logging/__init__.py", line 368, in getMessage
    msg = msg % self.args
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/py_info.py", line 270, in __str__
    (
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/py_info.py", line 297, in spec
    # this method is not used by itself, so here and called functions can import stuff locally
AttributeError: 'PythonInfo' object has no attribute 'free_threaded'
Call stack:
  File "/usr/local/lib/python3.10/threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 83, in _worker
    work_item.run()
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 274, in _execute_operation
    result = self._do_execute_operation(operation)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 385, in _do_execute_operation
    result: int = getattr(self, f"_execute_{method}")(operation)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 509, in _execute_install
    status_code = self._install(operation)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 543, in _install
    archive = self._download_link(operation, Link(package.source_url))
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/executor.py", line 714, in _download_link
    archive = self._chef.prepare(archive, output_dir=original_archive.parent)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/chef.py", line 43, in prepare
    return self._prepare_sdist(archive, destination=output_dir)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/chef.py", line 88, in _prepare_sdist
    return self._prepare(
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/installation/chef.py", line 49, in _prepare
    with isolated_builder(
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/utils/isolated_build.py", line 180, in isolated_builder
    with ephemeral_environment(
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/utils/env/__init__.py", line 47, in ephemeral_environment
    EnvManager.build_venv(
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/utils/env/env_manager.py", line 552, in build_venv
    cli_result = virtualenv.cli_run(args, setup_logging=False)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/run/__init__.py", line 31, in cli_run
    of_session = session_via_cli(args, options, setup_logging, env)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/run/__init__.py", line 49, in session_via_cli
    parser, elements = build_parser(args, options, setup_logging, env)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/run/__init__.py", line 77, in build_parser
    parser._interpreter = interpreter = discover.interpreter  # noqa: SLF001
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/discover.py", line 41, in interpreter
    self._interpreter = self.run()
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py", line 60, in run
    result = get_interpreter(python_spec, self.try_first_with, self.app_data, self._env)
  File "/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py", line 81, in get_interpreter
    LOGGER.info("proposed %s", interpreter)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1477, in info
    self._log(INFO, msg, args, **kwargs)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1624, in _log
    self.handle(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1634, in handle
    self.callHandlers(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1696, in callHandlers
    hdlr.handle(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 968, in handle
    self.emit(record)
  File "/myapp/.venv/lib/python3.10/site-packages/poetry/console/logging/io_handler.py", line 30, in emit
    self.handleError(record)
Message: '<fg=blue>proposed %s</>'
Arguments: (PythonInfo({'architecture': 64, 'base_exec_prefix': '/usr/local', 'base_prefix': '/usr/local', 'distutils_install': {'data': '', 'headers': 'include/python3.10/UNKNOWN', 'platlib': 'lib/python3.10/site-packages', 'purelib': 'lib/python3.10/site-packages', 'scripts': 'bin'}, 'exec_prefix': '/usr/local', 'executable': '/usr/local/bin/python3.10', 'file_system_encoding': 'utf-8', 'has_venv': True, 'implementation': 'CPython', 'max_size': 9223372036854775807, 'original_executable': '/usr/local/bin/python3.10', 'os': 'posix', 'path': ['/myapp/.venv/lib/python3.10/site-packages/virtualenv/discovery', '/usr/local/lib/python310.zip', '/usr/local/lib/python3.10', '/usr/local/lib/python3.10/lib-dynload', '/usr/local/lib/python3.10/site-packages'], 'platform': 'linux', 'prefix': '/usr/local', 'real_prefix': None, 'stdout_encoding': 'utf-8', 'sysconfig': {'makefile_filename': '/usr/local/lib/python3.10/config-3.10-x86_64-linux-gnu/Makefile'}, 'sysconfig_paths': {'data': '{base}', 'include': '{installed_base}/include/python{py_version_short}{abiflags}', 'platlib': '{platbase}/{platlibdir}/python{py_version_short}/site-packages', 'platstdlib': '{platbase}/{platlibdir}/python{py_version_short}', 'purelib': '{base}/lib/python{py_version_short}/site-packages', 'scripts': '{base}/bin', 'stdlib': '{installed_base}/{platlibdir}/python{py_version_short}'}, 'sysconfig_scheme': None, 'sysconfig_vars': {'PYTHONFRAMEWORK': '', 'abiflags': '', 'base': '/usr/local', 'installed_base': '/usr/local', 'platbase': '/usr/local', 'platlibdir': 'lib', 'py_version_short': '3.10'}, 'system_executable': '/usr/local/bin/python3.10', 'system_stdlib': '/usr/local/lib/python3.10', 'system_stdlib_platform': '/usr/local/lib/python3.10', 'version': '3.10.16 (main, Jan 14 2025, 02:25:17) [GCC 12.2.0]', 'version_info': VersionInfo(major=3, minor=10, micro=16, releaselevel='final', serial=0), 'version_nodot': '310'}),)

  Stack trace:

  19  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:274 in _execute_operation
        result = self._do_execute_operation(operation)

  18  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:385 in _do_execute_operation
        result: int = getattr(self, f"_execute_{method}")(operation)

  17  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:509 in _execute_install
        status_code = self._install(operation)

  16  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:543 in _install
        archive = self._download_link(operation, Link(package.source_url))

  15  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:714 in _download_link
        archive = self._chef.prepare(archive, output_dir=original_archive.parent)

  14  .venv/lib/python3.10/site-packages/poetry/installation/chef.py:43 in prepare
        return self._prepare_sdist(archive, destination=output_dir)

  13  .venv/lib/python3.10/site-packages/poetry/installation/chef.py:88 in _prepare_sdist
        return self._prepare(

  12  .venv/lib/python3.10/site-packages/poetry/installation/chef.py:49 in _prepare
        with isolated_builder(

  11  /usr/local/lib/python3.10/contextlib.py:135 in __enter__
        return next(self.gen)

  10  .venv/lib/python3.10/site-packages/poetry/utils/isolated_build.py:180 in isolated_builder
        with ephemeral_environment(

   9  /usr/local/lib/python3.10/contextlib.py:135 in __enter__
        return next(self.gen)

   8  .venv/lib/python3.10/site-packages/poetry/utils/env/__init__.py:47 in ephemeral_environment
        EnvManager.build_venv(

   7  .venv/lib/python3.10/site-packages/poetry/utils/env/env_manager.py:552 in build_venv
        cli_result = virtualenv.cli_run(args, setup_logging=False)

   6  .venv/lib/python3.10/site-packages/virtualenv/run/__init__.py:31 in cli_run
        of_session = session_via_cli(args, options, setup_logging, env)

   5  .venv/lib/python3.10/site-packages/virtualenv/run/__init__.py:49 in session_via_cli
        parser, elements = build_parser(args, options, setup_logging, env)

   4  .venv/lib/python3.10/site-packages/virtualenv/run/__init__.py:77 in build_parser
        parser._interpreter = interpreter = discover.interpreter  # noqa: SLF001

   3  .venv/lib/python3.10/site-packages/virtualenv/discovery/discover.py:41 in interpreter
        self._interpreter = self.run()

   2  .venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py:60 in run
        result = get_interpreter(python_spec, self.try_first_with, self.app_data, self._env)

   1  .venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py:82 in get_interpreter
        if interpreter.satisfies(spec, impl_must_match):

  AttributeError

  'PythonSpec' object has no attribute 'free_threaded'

  at .venv/lib/python3.10/site-packages/virtualenv/discovery/py_info.py:335 in satisfies
      331│                 return False
      332│         return True
      333│
      334│     _current_system = None
    → 335│     _current = None
      336│
      337│     @classmethod
      338│     def current(cls, app_data=None):
      339│         """

Cannot install fqdn.


  Stack trace:

  19  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:274 in _execute_operation
        result = self._do_execute_operation(operation)

  18  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:385 in _do_execute_operation
        result: int = getattr(self, f"_execute_{method}")(operation)

  17  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:509 in _execute_install
        status_code = self._install(operation)

  16  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:543 in _install
        archive = self._download_link(operation, Link(package.source_url))

  15  .venv/lib/python3.10/site-packages/poetry/installation/executor.py:714 in _download_link
        archive = self._chef.prepare(archive, output_dir=original_archive.parent)

  14  .venv/lib/python3.10/site-packages/poetry/installation/chef.py:43 in prepare
        return self._prepare_sdist(archive, destination=output_dir)

  13  .venv/lib/python3.10/site-packages/poetry/installation/chef.py:88 in _prepare_sdist
        return self._prepare(

  12  .venv/lib/python3.10/site-packages/poetry/installation/chef.py:49 in _prepare
        with isolated_builder(

  11  /usr/local/lib/python3.10/contextlib.py:135 in __enter__
        return next(self.gen)

  10  .venv/lib/python3.10/site-packages/poetry/utils/isolated_build.py:180 in isolated_builder
        with ephemeral_environment(

   9  /usr/local/lib/python3.10/contextlib.py:135 in __enter__
        return next(self.gen)

   8  .venv/lib/python3.10/site-packages/poetry/utils/env/__init__.py:47 in ephemeral_environment
        EnvManager.build_venv(

   7  .venv/lib/python3.10/site-packages/poetry/utils/env/env_manager.py:552 in build_venv
        cli_result = virtualenv.cli_run(args, setup_logging=False)

   6  .venv/lib/python3.10/site-packages/virtualenv/run/__init__.py:31 in cli_run
        of_session = session_via_cli(args, options, setup_logging, env)

   5  .venv/lib/python3.10/site-packages/virtualenv/run/__init__.py:49 in session_via_cli
        parser, elements = build_parser(args, options, setup_logging, env)

   4  .venv/lib/python3.10/site-packages/virtualenv/run/__init__.py:77 in build_parser
        parser._interpreter = interpreter = discover.interpreter  # noqa: SLF001

   3  .venv/lib/python3.10/site-packages/virtualenv/discovery/discover.py:41 in interpreter
        self._interpreter = self.run()

   2  .venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py:60 in run
        result = get_interpreter(python_spec, self.try_first_with, self.app_data, self._env)

   1  .venv/lib/python3.10/site-packages/virtualenv/discovery/builtin.py:82 in get_interpreter
        if interpreter.satisfies(spec, impl_must_match):

  AttributeError

  'PythonSpec' object has no attribute 'free_threaded'

  at .venv/lib/python3.10/site-packages/virtualenv/discovery/py_info.py:335 in satisfies
      331│                 return False
      332│         return True
      333│
      334│     _current_system = None
    → 335│     _current = None
      336│
      337│     @classmethod
      338│     def current(cls, app_data=None):
      339│         """

Cannot install isoduration.
